src/IPC/IPC.h

```diff
@@ -40,12 +40,16 @@ class PipeRx: public ReceiverIPC {
 };
 
 class QueueRx: public ReceiverIPC {
+    std::fstream *writeFile;
+    int qid;
```

Comment:
Maybe put those variables in private ? 
I think this is safer because then only this class can access to it.